### PR TITLE
ROX-12982: Drop custom SCCs [WIP]

### PR DESCRIPTION
**Pull request by vikin91**
## Description

All optional custom SCCs which names start with `stackrox-` should not be created anymore. Instead, our deployments should use the built-in SCCs.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

### Currently used SCCs

Ideally, all deployments shall use `restricted-v2` unless they have a reason for using more powerful SCCs.

| Component | Custom SCC before | Stock SCC before | SCC now | SCC now optimal? |
|--------|--------|--------|--------|--------|
| Sensor | `stackrox-sensor` | `restricted-v2` | `restricted-v2` | ✅ |
| Admission Controller | `stackrox-admission-control` | `restricted-v2` | `restricted-v2` | ✅ |
| Central | `stackrox-central` | `anyuid` | `anyuid` | ❌  |
| Central-db | `stackrox-central-db` | `anyuid` | `anyuid` | ❌  |
| Scanner | `stackrox-scanner` | `anyuid` | `anyuid` | ❌  |
| Scanner-db | `stackrox-scanner` | `anyuid` | `anyuid` | ❌  |
| Collector | `stackrox-collector` | `privileged` | `privileged` | ✅ |
| monitoring | `stackrox-monitoring` | - (always used custom SCC) | `nonroot-v2` | ❌ |
| stackrox-monitoring-kube-state-metrics | `?` | `?` | `nonroot-v2` | ❌  |

For all _SCC now optimal_==:x:, we should check if the container can run with `restricted-v2`.

### Here I tell how I validated my change

- [x] CI unit tests
- [ ] CI e2e tests
- [x] Manual deployment of OpenShift 4 cluster with monitoring and ensuring that all services are starting and working properly
- [ ] Operator deployment to OpenShift 4 cluster - confirming that all works and the option to install with custom SCCs is gone
- [ ] Upgrade tests - Helm
- [ ] Upgrade tests - Operator
- [ ] Following manual upgrade instructions for manifest-based installation 
- [ ] Check for interrelation with recent patch for sensor monitoring https://github.com/stackrox/stackrox/pull/7726
- [ ] Test with `ROX_OPENSHIFT_VERSION` unset or set to something different than `4`

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by ludydoo** [comment]
Looking at this, it seems like (some of) the SCCs are not even in use. For example, the `scanner` `Role` doesn't have the `use` permission for the `scanner-scc`

File: operator/apis/platform/v1alpha1/central_types.go [hunk 1]
```diff
@@ -53,10 +53,6 @@ type CentralSpec struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName=Customizations,order=6,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
 	Customize *CustomizeSpec `json:"customize,omitempty"`
 
-	// Miscellaneous settings.
-	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName=Miscellaneous,order=7,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
-	Misc *MiscSpec `json:"misc,omitempty"`
```
Comment (ludydoo) [inline operator/apis/platform/v1alpha1/central_types.go:58]: Never remove a property from a CRD !!! Must be marked as deprecated

File: operator/apis/platform/v1alpha1/common_types.go [hunk 2]
```diff
@@ -7,14 +7,6 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// MiscSpec defines miscellaneous settings for custom resources.
-type MiscSpec struct {
```
Comment (ludydoo) [inline operator/apis/platform/v1alpha1/common_types.go:11]: Same here

File: operator/apis/platform/v1alpha1/securedcluster_types.go [hunk 3]
```diff
@@ -80,10 +80,6 @@ type SecuredClusterSpec struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName=Customizations,order=10,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
 	Customize *CustomizeSpec `json:"customize,omitempty"`
 
-	// Miscellaneous settings.
-	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName=Miscellaneous,order=11,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
-	Misc *MiscSpec `json:"misc,omitempty"`
```
Comment (ludydoo) [inline operator/apis/platform/v1alpha1/securedcluster_types.go:85]: Same here

**Review by ludydoo — CHANGES_REQUESTED** [review]

File: operator/apis/platform/v1alpha1/central_types.go (hunk 1 above)
Comment (vikin91) [reply operator/apis/platform/v1alpha1/central_types.go:58]: Thanks! This ticket is parked for now, but when I come back to working on it, I will definitely fix that issue.

**Comment by vikin91** [comment]
This PR has been replaced by: https://github.com/stackrox/stackrox/pull/8663

@Maddosaurus please close this PR when you don't need it anymore.

**Pull request closed without merging**